static/src/javascripts/projects/common/modules/commercial/fiv-banner.js

```diff
@@ -40,6 +40,21 @@ const defaultEngagementBannerParams = () => ({
 const componentId = 'onemillion_fiv';
 
 const getBannerHtml = (params: EngagementBannerParams) => {
+    // hack to decrement, that makes this function not a pure function
```

Comment:
is `showBanner` not a better place to put this? inside the `if (messageShown)` block? since that function is side-effecting anyway and it makes it very clear that this is something you want to do each time the message is actually shown

Reply:
no I checked with @tomrf1 and the combined banner doesn't call showBanner, it just called getBannerHtml